ehzhang/HELPq
# Add Slack Integration for Tickets

See : #20 

## What in the world...

This PR provides the basic Slack integration as described in #20 which:
(1) Allows users to create/cancel/get tickets via Slack
(2) Posts status changes into a "mentor" Slack channel

## Uh, why?

We (the ops team) used this at [Swamphacks 2017](http://2017.swamphacks.com) with reasonable success; a little over 50% of our tickets were created via Slack. We (the mentors) really appreciated having a feed of new tickets come in. It was very easy to discuss them and tag those who could help out with them.

## Oh, ok, well, uh, how do I test this?

You can test this by following the new instructions in the README to set this code up. Then, go ahead and do the following while verifying the incoming webhook works:

1. Create a ticket via Slack (should succeed)
2. Cancel a ticket via Slack (should succeed)
3. Get a ticket via Slack (should fail; no ticket open!)
4. Cancel a ticket via Slack (should fail; no ticket open!)
5. Create a ticket via Slack (should succeed)
6. Claim the ticket via HELPq & test releasing and completing it
7. Create a ticket via HELPq and do (6) again

## Oh snap, this is cool, but I'd love for you to do [x]

Just ask! More than happy to add more functionality if you'd like. This is (notably) missing the ability for (1) users to rate tickets and (2) mentors to manage tickets.

Addressing (1), I couldn't really find a good way to _force_ users to have users rate tickets like HELPq does. Also, I'm not too keen on rating mentors since they're volunteering their time, but that's less of an excuse. :)

Addressing (2), I figured that mentors would be expected to have HELPq open so building them slash commands seemed like kind of a waste.

## Wait, do you even know how to use Meteor?

No. No I do not. If there's any coding standards I'm violating I'm very sorry! I just learned the minimum needed to do this. Just let me know what needs to be changed and I'll be more than happy to do it!
